### PR TITLE
PAINTROID-334 Refactor HSVColorPickerView to Kotlin

### PR DESCRIPTION
**Pull request by CodeChamp-SS**
The file that was refactored: HSVColorPickerView

https://jira.catrob.at/browse/PAINTROID-334

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: colorpicker/src/main/java/org/catrobat/paintroid/colorpicker/HSVColorPickerView.kt [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+import kotlin.math.min
+
+private const val BORDER_WIDTH_PX = 1f
+private const val ALPHA_PANEL_HEIGHT = 21f
+private const val HUE_PANEL_WIDTH = 30f
+private const val PANEL_SPACING = 10f
+private const val PALETTE_CIRCLE_TRACKING_RADIUS = 5f
+private const val RECTANGLE_TRACKER_OFFSET = 2f
+private const val HUE = 360f
+private const val CONSTANT_1 = 3
```
Comment (ldirry) [inline colorpicker/src/main/java/org/catrobat/paintroid/colorpicker/HSVColorPickerView.kt:46]: Constant_1 should be the channels amount
and Constant_2 should be the max alpha value

File: colorpicker/src/main/java/org/catrobat/paintroid/colorpicker/HSVColorPickerView.kt [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    private var drawingOffset = 0f
+    private var satValPaint: Paint
+    private var satValTrackerPaint: Paint
+    private var huePaint: Paint
+    private var hueTrackerPaint: Paint
+    private var alphaPaint: Paint
+    private var borderPaint: Paint
+    private var checkeredPaint: Paint
+    private var onColorChangedListener: OnColorChangedListener? = null
+    private var drawingRect: RectF? = null
```
Comment (ldirry) [inline colorpicker/src/main/java/org/catrobat/paintroid/colorpicker/HSVColorPickerView.kt:71]: all the rects get initialized in the init block, and they never get set to null aftwards, so I think they can be a non nullable val

**Review by ldirry — CHANGES_REQUESTED** [review]
nicely done, just two small remarks :)

File: colorpicker/src/main/java/org/catrobat/paintroid/colorpicker/HSVColorPickerView.kt (hunk 2 above)
Comment (CodeChamp-SS) [reply colorpicker/src/main/java/org/catrobat/paintroid/colorpicker/HSVColorPickerView.kt:71]: yeah, nice catch ;D

**Comment by CodeChamp-SS** [comment]
> nicely done, just two small remarks :)

made those changes.

**Review by ldirry — APPROVED** [review]

**Pull request merged**